src/mxbase.jl

```diff
@@ -1,104 +1,73 @@
+    end
+    matlab_lib_path = joinpath(matlab_home, "bin", matlab_lib_dir)
+    if !isdir(matlab_lib_path)
+        error("The MATLAB library path could not be found.")
+    end
+    return matlab_lib_path
+end
 
+function matlab_startcmd()
+    matlab_home = matlab_home_path()
     if !is_windows()
-        default_startcmd = joinpath(matlab_homepath, "bin", "matlab")
+        default_startcmd = joinpath(matlab_home, "bin", "matlab")
         if !isfile(default_startcmd)
             error("The MATLAB path is invalid. Set the MATLAB_HOME evironmental variable to the MATLAB root.")
         end
-        default_startcmd = "exec $(Base.shell_escape(default_startcmd)) -nosplash"
+        default_startcmd = "exec $(Base.shell_escape(default_startcmd))"
     elseif is_windows()
-        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.WORD_SIZE == 32 ? "win32" : "win64"), "MATLAB.exe")
+        default_startcmd = joinpath(matlab_home, "bin", (Sys.WORD_SIZE == 32 ? "win32" : "win64"), "MATLAB.exe")
         if !isfile(default_startcmd)
             error("The MATLAB path is invalid. Set the MATLAB_HOME evironmental variable to the MATLAB root.")
         end
-        default_startcmd *= " -nosplash"
-    end
-
-    # Get path to MATLAB libraries
-    if is_linux()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "glnx86" : "glnxa64"
-    elseif is_apple()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "maci" : "maci64"
-    elseif is_windows()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "win32" : "win64"
-    end
-    matlab_library_path = joinpath(matlab_homepath, "bin", matlab_library_dir)
-
-    if matlab_library_path != nothing && !isdir(matlab_library_path)
-        matlab_library_path = nothing
-    end
-end
-get_paths()
-
-matlab_library(lib::String) = matlab_library_path == nothing ? lib : joinpath(matlab_library_path, lib)
-
-# libmx (loaded when the module is imported)
-
-function load_libmx()
-    global libmx
-    if libmx == C_NULL
-        libmx = dlopen(matlab_library("libmx"), RTLD_GLOBAL | RTLD_LAZY)
-        libmx == C_NULL && error("Failed to load libmx.")
-    end
-end
-load_libmx()
-
-# libmat (loaded when the module is imported)
-
-function load_libmat()
-    global libmat
-    if libmat == C_NULL
-        libmat = dlopen(matlab_library("libmat"), RTLD_GLOBAL | RTLD_LAZY)
-        libmat == C_NULL && error("Failed to load libmat.")
     end
+    return default_startcmd
 end
-load_libmat()
-
-# libeng (loaded when needed)
 
-function load_libeng()
-    global libeng
-    if libeng == C_NULL
-        libeng = dlopen(matlab_library("libeng"), RTLD_GLOBAL | RTLD_LAZY)
-        libeng == C_NULL && error("Failed to load libeng.")
-    end
-end
+# helper library access function
 
-engfunc(fun::Symbol) = dlsym(libeng::Ptr{Void}, fun)
-mxfunc(fun::Symbol) = dlsym(libmx::Ptr{Void}, fun)
-matfunc(fun::Symbol) = dlsym(libmat::Ptr{Void}, fun)
+engfunc(fun::Symbol) = Libdl.dlsym_e(libeng[], fun)
```

Comment:
Unless there's absolutely no way that these symbols are missing (in which case there's also no reason to change this) it's better to give an error about missing symbol rather than a `UndefRefError`.